hoomd/util.py

```diff
@@ -37,7 +38,9 @@ def is_iterable(obj):
 
 def bad_iterable_type(obj):
     '''Returns True if str or dict.'''
-    return isinstance(obj, str) or isinstance(obj, dict)
+    return isinstance(
+        obj,
+        (str, dict, io.TextIOBase, io.BufferedIOBase, io.RawIOBase, io.IOBase))
```

Comment:
`io.TextIOBase, io.BufferedIOBase, io.RawIOBase` are already subclasses of `io.IOBase`.
```suggestion
        (str, dict, io.IOBase))
```